### DISCUSSION TITLE
Feature Request: Allow for setting headers in httpfs

### DISCUSSION THREAD

**Discussion by AndrewLWilson** [duckdb/duckdb#9233] [Ideas]
Why do you want this feature?

Most http(s) APIs require some type of authorization, and sometimes that authorization requires specific headers to be included.  For example, [Metabase](https://github.com/metabase/metabase) requires an 'X-Metabase-Session' header be included or it returns a 401.

Being able to load json responses directly into duckdb is such an amazing feature, and its real-world usefulness would greatly increase if we could set headers for auth purposes!

It could be additional configs to be set a la s3_access_key_id.  Possibly better would be something like:
`select * from request('https://api.spacexdata.com/v3/capsules', 'spacex-key: xxxxxxx')`;

**Comment by Mause** [comment]
Duplicate of https://github.com/duckdb/duckdb/discussions/6782
